### PR TITLE
feat: add logger to show badges

### PR DESCRIPTION
**Pull request by JellyBellyDev**
Example of log:
```
badge-poser-phpfpm | [2021-05-18T11:02:56.155152+00:00] badges.INFO: /phpunit/phpunit/v {"badgeData":{"subject":"stable","status":"9.5.4","color":"#28a3df","format":"svg"},"requestHeaders":{"cookie":["_ga=GA1.2.1221247998.1617793280; _gid=GA1.2.1449274797.1621268707"],"accept-language":["it-IT,it;q=0.9,en-US;q=0.8,en;q=0.7,de;q=0.6,es;q=0.5,fr;q=0.4,ru;q=0.3,da;q=0.2,pt;q=0.1"],"accept-encoding":["gzip, deflate"],"accept":["text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9"],"user-agent":["Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36"],"upgrade-insecure-requests":["1"],"cache-control":["no-cache"],"pragma":["no-cache"],"connection":["keep-alive"],"host":["poser.local:8001"],"content-length":[""],"content-type":[""],"x-php-ob-level":["1"]},"responseHeaders":{"cache-control":["max-age=3600, public, s-maxage=3600"],"date":["Tue, 18 May 2021 11:02:56 GMT"],"content-type":["image/svg+xml;charset=utf-8"],"content-disposition":["inline; filename=stable-954-28a3dfsvg.svg"]}} []
```

Content of json:
```json
{
   "badgeData":{
      "subject":"stable",
      "status":"9.5.4",
      "color":"#28a3df",
      "format":"svg"
   },
   "requestHeaders":{
      "cookie":[
         "_ga=GA1.2.1221247998.1617793280; _gid=GA1.2.1449274797.1621268707"
      ],
      "accept-language":[
         "it-IT,it;q=0.9,en-US;q=0.8,en;q=0.7,de;q=0.6,es;q=0.5,fr;q=0.4,ru;q=0.3,da;q=0.2,pt;q=0.1"
      ],
      "accept-encoding":[
         "gzip, deflate"
      ],
      "accept":[
         "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9"
      ],
      "user-agent":[
         "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36"
      ],
      "upgrade-insecure-requests":[
         "1"
      ],
      "cache-control":[
         "no-cache"
      ],
      "pragma":[
         "no-cache"
      ],
      "connection":[
         "keep-alive"
      ],
      "host":[
         "poser.local:8001"
      ],
      "content-length":[
         ""
      ],
      "content-type":[
         ""
      ],
      "x-php-ob-level":[
         "1"
      ]
   },
   "responseHeaders":{
      "cache-control":[
         "max-age=3600, public, s-maxage=3600"
      ],
      "date":[
         "Tue, 18 May 2021 11:02:56 GMT"
      ],
      "content-type":[
         "image/svg+xml;charset=utf-8"
      ],
      "content-disposition":[
         "inline; filename=stable-954-28a3dfsvg.svg"
      ]
   }
}
```

Do you think it might be useful? Is there anything useless to take off?

### CODE REVIEW

**Review by AlessandroMinoccheri — APPROVED** [review]
LGTM at the moment 👍

**Review by AlessandroMinoccheri — COMMENTED** [review]
What do you think about adding some tests?

**Comment by JellyBellyDev** [comment]
> What do you think about adding some tests?

Obviously, I wanted to wait for the output to be as requested and then proceed with adding them!

**Review by garak — APPROVED** [review]

File: src/Badge/Service/ImageFactory.php [hunk 1]
```diff
@@ -4,19 +4,24 @@
...
 {
-    public function __construct(private Poser $generator)
+    public function __construct(private Poser $generator, private EventDispatcherInterface $eventDispatcher)
     {
     }
 
     public function createFromBadge(BadgeInterface $badge): Image
     {
+        $badgeEvent = new BadgeEvent($badge);
+        $this->eventDispatcher->dispatch($badgeEvent, BadgeEvent::NAME);
```
Comment (garak) [inline src/Badge/Service/ImageFactory.php:23]: Not critical, but dispatching events with name is not used anymore. You can remove second argument (and remove constant `NAME` from event, too)

Comment (JellyBellyDev) [reply src/Badge/Service/ImageFactory.php]: Nice catch! Fixed! ;)

**Comment by fabiocicerchia** [comment]
LGTM :+1:

**Review by fabiocicerchia — APPROVED** [review]

**Comment by JellyBellyDev** [comment]
I add some tests! ;)

**Pull request merged**